### ISSUE TITLE
Bunch of sigsegvs when adding users to teams

### ISSUE THREAD

**Issue by sokoow** [mattermost/platform#5744]
#### Summary
```
Version: 3.6.0
Build Number: 3.6.2
Build Date: Tue Jan 31 21:02:59 UTC 2017
Build Hash: 1a9891f0f5671551d28be54a99155b907480cc5c
Build Enterprise Ready: true
```
#### Steps to reproduce
Not entirely sure, but getting this when adding a user to the team:

```
/opt/mattermost/bin/platform team add TEAM USER
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x416488]

goroutine 1 [running]:
panic(0xc9dae0, 0xc4200120e0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
main.getTeamFromTeamArg(0x7fff4d11c84f, 0x6, 0x0)
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/cmd/platform/teamargs.go:21 +0x28
main.addUsersCmdF(0x119d040, 0xc4200113e0, 0x2, 0x2, 0x0, 0x0)
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/cmd/platform/team.go:141 +0x6f
github.com/mattermost/platform/vendor/github.com/spf13/cobra.(*Command).execute(0x119d040, 0xc420011240, 0x2, 0x2, 0x119d040, 0xc420011240)
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/vendor/github.com/spf13/cobra/command.go:632 +0x23e
github.com/mattermost/platform/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4200acb40, 0x0, 0x0, 0x0)
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/vendor/github.com/spf13/cobra/command.go:722 +0x367
github.com/mattermost/platform/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4200acb40, 0xc4201572f0, 0xb)
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/vendor/github.com/spf13/cobra/command.go:681 +0x2b
main.main()
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/cmd/platform/mattermost.go:50 +0x49f
```
and now also:

```
/opt/mattermost/bin/platform version
Version: 3.6.0
Build Number: 3.6.2
Build Date: Tue Jan 31 21:02:59 UTC 2017
Build Hash: 1a9891f0f5671551d28be54a99155b907480cc5c
Build Enterprise Ready: true
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x41a6e8]

goroutine 1 [running]:
panic(0xc9dae0, 0xc4200100e0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
main.printVersion()
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/cmd/platform/version.go:29 +0x428
main.versionCmdF(0x119f020, 0x11c5cb0, 0x0, 0x0)
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/cmd/platform/version.go:20 +0x30
github.com/mattermost/platform/vendor/github.com/spf13/cobra.(*Command).execute(0x119f020, 0x11c5cb0, 0x0, 0x0, 0x119f020, 0x11c5cb0)
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/vendor/github.com/spf13/cobra/command.go:636 +0x443
github.com/mattermost/platform/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42019a480, 0x0, 0x0, 0x0)
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/vendor/github.com/spf13/cobra/command.go:722 +0x367
github.com/mattermost/platform/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42019a480, 0xc4200113a0, 0xb)
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/vendor/github.com/spf13/cobra/command.go:681 +0x2b
main.main()
	/home/ubuntu/workspace/msr/enterprise-release/src/github.com/mattermost/platform/cmd/platform/mattermost.go:50 +0x49f

```

#### Expected behavior
No sigsegvs

**Comment by lindy65** [comment]
Hi @sokoow 

Thanks for your issue report!

Do you still have the same issue if you update to [Mattermost v3.7.3](https://about.mattermost.com/download/)  (released March 16th)?

**Comment by jasonblais** [comment]
@sokoow Can you also try running the command inside the folder `/var/opt/gitlab/mattermost/bin`, if you haven't yet?

**Comment by mattiasb** [comment]
I have the exact same issue (with `gitlab-ce-9.0.1-ce.0.el6.x86_64` omnibus on CentOS 6.5).
Running the command inside the `/var/opt/gitlab/mattermost/bin` folder makes no difference, I get the same SIGSEGV as @sokoow above.

One issue though is that it seems like GitLab needs to issue another update for us to be able to upgrade.

**Comment by lindy65** [comment]
Thanks for your note @mattiasb,

Could we have your help [opening an issue](https://gitlab.com/gitlab-org/omnibus-gitlab/issues) with GitLab requesting an update?

**Comment by mattiasb** [comment]
Sure. However, do you have experience telling that 3.7.3 will solve the issue?

**Comment by mattiasb** [comment]
Posted [issue 2169](https://gitlab.com/gitlab-org/omnibus-gitlab/issues/2169) on the GitLab issue tracker.

**Comment by lindy65** [comment]
Hi @mattiasb,

Thanks for opening an issue with GitLab.

Difficult to say whether upgrading to Mattermost v3.7.3 will solve your issue as you're using GitLab omnibus whereas Sokoow isn't.  Perhaps you can let us know?

**Comment by mattiasb** [comment]
Hm, let me rephrase. Did you make an attempt at a fix for an issue that is similar to what me and Sokoow is experiencing?

**Comment by lindy65** [comment]
Apologies @mattiasb ! I misunderstood your question...

No, we did not make a specific fix for the issue you are experiencing.  The suggestion to Sokoow to upgrade to the latest version was because he was experiencing the problem in an old version of Mattermost which might not reproduce on the latest version.

**Comment by sokoow** [comment]
Sorry, I didn't get a chance to upgrade my install yet - is it a drop in install ? is there any risk of upgrading in place ?

**Comment by mattiasb** [comment]
@lindy65 ok gotcha!

**Comment by lindy65** [comment]
Thanks @mattiasb!

@sokoow, upgrading from v3.6.0 to v3.7.3 should be smooth and you can follow the upgrade guide [here](https://docs.mattermost.com/administration/upgrade.html#upgrade-guide)

Does this help?

**Comment by sokoow** [comment]
ok, upgrading to 3.7.3 helped, but now I have a different problem, all my users profile pictures are gone! upgrade guide doesn't include instructions on how to port custom emoticons and profile pictures, help please!

**Comment by sokoow** [comment]
another one:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xcc1028]

goroutine 1 [running]:
main.getUserFromUserArg(0x7ffcfbf9a8b2, 0x5, 0x1)
	/var/lib/jenkins/jobs/msr/jobs/d/jobs/enterprise-release/workspace/src/github.com/mattermost/platform/cmd/platform/userargs.go:21 +0x28
main.getUsersFromUserArgs(0xc420158c80, 0x1, 0x1, 0xb, 0x6, 0xe5cc03)
	/var/lib/jenkins/jobs/msr/jobs/d/jobs/enterprise-release/workspace/src/github.com/mattermost/platform/cmd/platform/userargs.go:13 +0xa2
main.changeUsersActiveStatus(0xc420158c80, 0x1, 0x1, 0x1)
	/var/lib/jenkins/jobs/msr/jobs/d/jobs/enterprise-release/workspace/src/github.com/mattermost/platform/cmd/platform/user.go:162 +0x43
main.userActivateCmdF(0x13022a0, 0xc420158c80, 0x1, 0x1, 0x0, 0x0)
	/var/lib/jenkins/jobs/msr/jobs/d/jobs/enterprise-release/workspace/src/github.com/mattermost/platform/cmd/platform/user.go:157 +0xc9
github.com/mattermost/platform/vendor/github.com/spf13/cobra.(*Command).execute(0x13022a0, 0xc420158c10, 0x1, 0x1, 0x13022a0, 0xc420158c10)
	/var/lib/jenkins/jobs/msr/jobs/d/jobs/enterprise-release/workspace/src/github.com/mattermost/platform/vendor/github.com/spf13/cobra/command.go:643 +0x3ef
github.com/mattermost/platform/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc421158000, 0x0, 0x0, 0x0)
	/var/lib/jenkins/jobs/msr/jobs/d/jobs/enterprise-release/workspace/src/github.com/mattermost/platform/vendor/github.com/spf13/cobra/command.go:734 +0x339
github.com/mattermost/platform/vendor/github.com/spf13/cobra.(*Command).Execute(0xc421158000, 0xc420158b90, 0xb)
	/var/lib/jenkins/jobs/msr/jobs/d/jobs/enterprise-release/workspace/src/github.com/mattermost/platform/vendor/github.com/spf13/cobra/command.go:692 +0x2b
main.main()
	/var/lib/jenkins/jobs/msr/jobs/d/jobs/enterprise-release/workspace/src/github.com/mattermost/platform/cmd/platform/mattermost.go:50 +0x4b2
```

**Comment by lindy65** [comment]
Hi @sokoow, 

There is [this post](https://forum.mattermost.org/t/anyone-missing-profile-pics-after-3-0-upgrade-while-using-local-storage/1546/7) which might help with the problem of missing profile pictures after upgrade.

Regarding the panic error you're getting, [this](https://forum.mattermost.org/t/solved-gitlab-8-9-panic-runtime-error-invalid-memory-address-or-nil-pointer-dereference/1810) might help?

Let us know how it goes?

**Comment by lindy65** [comment]
Hi @sokoow,

Just circling back to see whether you've managed to resolve your issue?

**Comment by sokoow** [comment]
We're good, activating/deactivating users is resolved in new version

**Comment by lindy65** [comment]
Awesome!  Thanks for the update @sokoow :)
